### PR TITLE
tests/shell: allow not using socat, do so on z1

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

For some reason, socat doesn't work properly on z1.

### CODE REVIEW

File: tests/shell/tests/01-run.py [hunk 1]
```diff
@@ -27,6 +27,9 @@
     '\t  \d | running  Q |   7'
 )
 
+RIOT_TERMINAL = os.environ.get('RIOT_TERMINAL')
+CLEANTERMS = { "cleanterm", "socat" }
```
Comment (fjmolinas) [inline tests/shell/tests/01-run.py:31]: `cleanterm` is used already for all tests, so it should just be socat

File: tests/shell/Makefile [hunk 2]
```diff
@@ -6,10 +6,17 @@ USEMODULE += shell
...
+ifeq (z1, $(BOARD))
+  RIOT_TERMINAL ?= pyterm
+endif
+
 # Use a terminal that does not introduce extra characters into the stream.
 RIOT_TERMINAL ?= socat
 
-DISABLE_MODULE += test_utils_interactive_sync
+# the test script skips tests if socat is not used
+include $(RIOTBASE)/makefiles/utils/variables.mk
```
Comment (fjmolinas) [inline tests/shell/Makefile:18]: Why is this needed it is included in, maybe move after `Makefile.include`?

https://github.com/RIOT-OS/RIOT/blob/06dca07b0a03dd034f85719a1cf7a5266d0aac86/Makefile.include#L143-L145

File: tests/shell/Makefile [hunk 3]
```diff
@@ -6,10 +6,17 @@ USEMODULE += shell
 USEMODULE += shell_commands
 USEMODULE += ps
 
+# for z1, socat doesn't work (unknown reason)
+ifeq (z1, $(BOARD))
+  RIOT_TERMINAL ?= pyterm
+endif
```
Comment (fjmolinas) [inline tests/shell/Makefile:12]: I agree with this workaround not worth spending time investigating an old non available `BOARD`

File: tests/shell/tests/01-run.py (hunk 1 above)
Comment (kaspar030) [reply tests/shell/tests/01-run.py:31]: fixed

Comment (fjmolinas) [reply tests/shell/tests/01-run.py:31]: Did you forget to push?

File: tests/shell/Makefile (hunk 2 above)
Comment (kaspar030) [reply tests/shell/Makefile]: > maybe move after `include $(RIOTBASE)/Makefile.include`?
> 
> [](https://github.com/RIOT-OS/RIOT/blob/06dca07b0a03dd034f85719a1cf7a5266d0aac86/Makefile.include#L143-L145)

that does the trick, fixed

**Comment by fjmolinas** [comment]
@kaspar030 can you address the styling issues travis points ou:

```
Running "./dist/tools/flake8/check.sh" x

Command output:

	There are style issues in the following Python scripts:

	

	tests/shell/tests/01-run.py:31:15: E201 whitespace after '{'

	tests/shell/tests/01-run.py:31:23: E202 whitespace before '}'

	tests/shell/tests/01-run.py:212:1: E305 expected 2 blank lines after class or function definition, found 1
```

**Comment by kaspar030** [comment]
> @kaspar030 can you address the styling issues travis points ou:

done and squashed

**Comment by kaspar030** [comment]
```
--- run_test job results (2 failed, 22 passed, 24 total):
    failed:
    tests/suit_manifest/nrf52dk:gnu
    tests/gnrc_ipv6_fwd_w_sub/nrf52dk:gnu
```

unrelated fails, will re-trigger.

**Comment by kaspar030** [comment]
thanks for reviewing!
